mnemosyne/repair.py

```
@@ -0,0 +1,1028 @@
+
+    descriptor: int | None = None
+    parent_fd: int | None = None
+    try:
+        parent_fd = _open_safe_directory_fd(path.parent)
+        flags = os.O_RDONLY
+        if hasattr(os, "O_NOFOLLOW"):
+            flags |= os.O_NOFOLLOW
+        descriptor = os.open(path.name, flags, dir_fd=parent_fd)
+        info = os.fstat(descriptor)
+    except (OSError, ValueError) as error:
+        raise RepairError("Database identity could not be verified") from error
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)